packages/block-library/src/button/editor.scss

```diff
@@ -79,3 +79,40 @@ div[data-type="core/button"] {
 .editor-styles-wrapper .wp-block-button[style*="text-decoration"] .wp-block-button__link {
 	text-decoration: inherit;
 }
+
+.editor-styles-wrapper .wp-block-button .wp-block-button__link {
```

Comment:
I'm not a huge fan of having to add this much CSS to provide defaults that can overcome styles in core. If anyone has any ideas on a better approach, I'd be happy to evolve this one.

Without these tweaks the user has to set multiple fields before a border shows up. Even now the default border color is `currentColor` so when the background is white and the button text is white, the border isn't really visible.

Reply:
It doesn't seem like we need to replicate this in `editor.scss` and `styles.scss` - if I removed this section from here the CSS from `styles.scss` seemed to be applied in the editor and the frontend?

Reply:
We do need to add this CSS to overcome [core's theme.json button element styles](https://github.com/WordPress/gutenberg/blob/d9da1250c77d3dc083729b72b8143dfc7bc222ba/lib/theme.json#L437) which reset `border-width` to `0`. 

The theme.json styles get the `.editor-styles-wrapper` class prefixed to their selectors in the editor. As a result, those styles override the styles in the frontend `style.scss`.

Without these styles, a user needs to adjust the border width before they can see any border. 

To witness this behaviour:
1. Delete or disable the styles here in the editor.scss file
2. Make sure you don't have any `core/button` styles in your theme.json
3. Edit a post and add a button
4. Select the button and select only a border color
5. Note that there is no visible border due to the core theme.json border width reset for button elements.



https://user-images.githubusercontent.com/60436221/217154597-f2d1ddb7-ed34-4cde-8cc5-2fdae382a5a3.mp4

Reply:
Thanks for the extra detail - I was adding the border width and then the color so wasn't picking this up. I have modified the comment to specify this detail like the one in style.scss does.